tests/js/spec/views/organizationGroupDetails/groupEventDetails.spec.tsx

```diff
@@ -0,0 +1,480 @@
+    url: `/issues/${group.id}/tags/`,
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/groups/${group.id}/integrations/`,
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/groups/${group.id}/external-issues/`,
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/issues/${group.id}/current-release/`,
+    body: {currentRelease: null},
+  });
+
+  MockApiClient.addMockResponse({
+    url: '/prompts-activity/',
+    body: undefined,
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/organizations/${organization.slug}/has-mobile-app-events/`,
+    body: null,
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/projects/${organization.slug}/${project.slug}/events/${event.id}/grouping-info/`,
+    body: {},
+  });
+  MockApiClient.addMockResponse({
+    url: `/projects/${organization.slug}/${project.slug}/codeowners/`,
+    body: [],
+  });
+  MockApiClient.addMockResponse({
+    url: `/organizations/${organization.slug}/code-mappings/`,
+    method: 'GET',
+    body: [],
+  });
+
+  // Sentry related mocks
+  MockApiClient.addMockResponse({
+    url: '/sentry-apps/',
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/organizations/${organization.slug}/sentry-apps/`,
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/organizations/${organization.slug}/sentry-app-installations/`,
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: `/organizations/${organization.slug}/sentry-app-components/?projectId=${project.id}`,
+    body: [],
+  });
+
+  MockApiClient.addMockResponse({
+    url: '/projects/org-slug/project-slug/',
+    body: project,
+  });
+};
+
+describe('groupEventDetails', () => {
+  beforeEach(() => {
+    MockApiClient.clearMockResponses();
+    CommitterStore.init();
+  });
+
+  afterEach(function () {
+    MockApiClient.clearMockResponses();
+    CommitterStore.teardown();
+    (browserHistory.replace as jest.Mock).mockClear();
+  });
+
+  it('redirects on switching to an invalid environment selection for event', async function () {
+    const props = makeDefaultMockData();
+    mockGroupApis(props.organization, props.project, props.group, props.event);
+
+    const {rerender} = render(<TestComponent {...props} />);
+    expect(browserHistory.replace).not.toHaveBeenCalled();
+
+    rerender(
+      <TestComponent environments={[{id: '1', name: 'prod', displayName: 'Prod'}]} />
+    );
+
+    await waitFor(() => expect(browserHistory.replace).toHaveBeenCalled());
+  });
+
+  it('does not redirect when switching to a valid environment selection for event', async function () {
+    const props = makeDefaultMockData();
+    mockGroupApis(props.organization, props.project, props.group, props.event);
+
+    const {rerender} = render(<TestComponent {...props} />);
+
+    expect(browserHistory.replace).not.toHaveBeenCalled();
+    rerender(<TestComponent environments={[]} />);
+
+    await act(async () => {
+      await tick();
```

Comment:
can we replace `await tick()` with some RTL async query?

Reply:
Yeah, I'll just check that the component renders for now